codegen/masm/src/codegen.rs

```diff
@@ -287,6 +365,140 @@ impl<'ast> CodeGenerator<'ast> {
         Ok(())
     }
 
+    /// Emits code for the procedure `get_exemptions_points`.
+    ///
+    /// The generated procedure contains the precomputed exeption points to be used when computing
+    /// the divisors. The values are returned in the stack.
+    fn gen_get_exemptions_points(&mut self) -> Result<(), CodegenError> {
+        // Notes:
+        // - Computing the exemption points on the fly would require 1 exponentiation to find the
+        // root-of-unity from the two-adacity, followed by another exponetiation to compute the
+        // two-to-last exemption point, and a multiplication to compute the last exemption point.
+        // Each exponentiation is 41 cycles, giving around 83 cycles to compute the values.
+        // - For the range from powers 3 to 32 there are 30 unique values, which requires 8 words
+        // of data. Storing the data to memory requires pushing the 4 elements of a word to the
```

Comment:
I think the ranges need to be update, either for the current approach or the other proposed one.